### DISCUSSION TITLE
Does %file have option to copy a (big) directory except a particular file in it?

### DISCUSSION THREAD

**Discussion by Heng-Zhou** [sylabs/singularity#3374] [Q&A] [answered]
I would like to use `%file` section in def file to copy a big directory into container, but for same reason a file in this directory is unreadable (thus uncopiable). This would cause the copy and the whole building process to exit. Does `%file` have any option to copy a (big) directory except a particular file in it? The Go `file.Match` syntax does not have exclusion pattern. Currently, I manually write files/sub-directories under the big directory but it is too tedious; I hope there is an exclusion option in `%file` section of def file that could make it easy. Thanks.

**Comment by dtrudg** [comment, marked as answer]
No. There is no exclusion option. As you noted, the `filepath.Match` syntax is used - so there is no negation / exclusion.
